caffe2/CMakeLists.txt

```diff
@@ -76,12 +76,38 @@ install(FILES ${PROJECT_BINARY_DIR}/caffe2/core/macros.h
 
 
 # ---[ List of libraries to link with
-
 add_library(caffe2_protos $<TARGET_OBJECTS:Caffe_PROTO> $<TARGET_OBJECTS:Caffe2_PROTO>)
 add_dependencies(caffe2_protos Caffe_PROTO Caffe2_PROTO)
 # TODO: enable using lite protobuf.
-target_link_libraries(caffe2_protos PUBLIC protobuf::libprotobuf)
+# If we are going to link protobuf locally inside caffe2 libraries, what we will do is
+# to create a helper static library that always contains libprotobuf source files, and
+# link the caffe2 related dependent libraries to it.
 target_include_directories(caffe2_protos INTERFACE $<INSTALL_INTERFACE:include>)
+# Reason for this public dependency is as follows:
+# (1) Strictly speaking, we should not expose any Protobuf related functions. We should
+#     only use function interfaces wrapped with our own public API, and link protobuf
+#     locally.
+# (2) However, currently across the Caffe2 codebase, we have extensive use of protobuf
```

Comment:
Is the python extension taken care of with these changes? Just checking.

Reply:
Currently no - it still depends on the transitive dependency of libprotobuf static. But no explicit dependency on any libprotobuf.so library is needed now.